lib/Convos/Controller/Notifications.pm

```diff
@@ -16,7 +16,12 @@ sub read {
   my $self = shift->openapi->valid_input or return;
   my $user = $self->backend->user        or return $self->reply->errors([], 401);
 
-  return $user->unread(0)->save_p->then(sub { $self->render(openapi => {}) });
+  Mojo::Promise->all($user->connections->map(sub {
+      my $conn = shift;
+      $conn->conversations->map( sub { shift->notifications(0); });
+      $conn->save_p();
+    })
+  )->then(sub { $self->render(openapi => {}) });
```

Comment:
I don't think you need this complexity, since the connection is saved on interval. (You should be rather unlucky if it doesn't get saved automatically) I suggest something like this:

```
$user->connections->map(sub { shift->conversations->map(notifications => 0) });
$self->render(openapi => {});
```

But the problem with the original code is that `then()` is called on the collection (return value from map) not the promise method all(). Pretty sure this would work:

```
Mojo::Promise->all(
  $user->connections->map(sub {
    my $conn = shift;
    $conn->conversations->map(sub { shift->notifications(0); });
    return $conn->save_p;
  })->each
)->then(sub { $self->render(openapi => {}) });
```

Reply:
Thanks! Revised as suggested